panic index out of range inserting empty []byte in BLOB column

- os Version: Windows 10
- cli-driver Version: 11.1
- Db2 Server Version: 11.5.0.1077
- go Version: 1.13.5
<details><summary><code>go env</code> Output:</summary><br><pre>
$ go env
set GO111MODULE=
set GOARCH=amd64
set GOBIN=
set GOCACHE=C:\Users\sdelacruz\AppData\Local\go-build
set GOENV=C:\Users\sdelacruz\AppData\Roaming\go\env
set GOEXE=.exe
set GOFLAGS=
set GOHOSTARCH=amd64
set GOHOSTOS=windows
set GONOPROXY=
set GONOSUMDB=
set GOOS=windows
set GOPATH=C:\Users\sdelacruz\Documents\go
set GOPRIVATE=
set GOPROXY=https://proxy.golang.org,direct
set GOROOT=c:\go
set GOSUMDB=sum.golang.org
set GOTMPDIR=
set GOTOOLDIR=c:\go\pkg\tool\windows_amd64
set GCCGO=gccgo
set AR=ar
set CC=gcc
set CXX=g++
set CGO_ENABLED=1
set GOMOD=C:\Users\sdelacruz\Documents\go\src\sdi\go.mod
set CGO_CFLAGS=-g -O2
set CGO_CPPFLAGS=
set CGO_CXXFLAGS=-g -O2
set CGO_FFLAGS=-g -O2
set CGO_LDFLAGS=-g -O2
set PKG_CONFIG=pkg-config
set GOGCCFLAGS=-m64 -mthreads -fmessage-length=0 -fdebug-prefix-map=C:\Users\SDELAC~1\AppData\Local\Temp\go-build670067514=/tmp/go-build -gno-record-gcc-switches
</pre></details>

The driver cannot insert nil in a column with data type BLOB, launch this error from IBM:
`SQLExecute: {HY099} [IBM][CLI Driver] CLI0164E  Nullable type out of range. SQLSTATE=HY099`

So trying to insert a empty []byte in a data type BLOB return this panic:
```
panic: runtime error: index out of range [0] with length 0

goroutine 18 [running]:
github.com/ibmdb/go_ibm_db.(*Parameter).BindValue(0xc0000940d8, 0x10001, 0x3, 0xcba240, 0xc00029a240, 0x0, 0x0)
        C:/Users/sdelacruz/Documents/go/pkg/mod/github.com/ibmdb/go_ibm_db@v0.2.0/param.go:126 +0x17a8
github.com/ibmdb/go_ibm_db.(*ODBCStmt).Exec(0xc00225e410, 0xc000322640, 0x14, 0x14, 0x14, 0x14)
        C:/Users/sdelacruz/Documents/go/pkg/mod/github.com/ibmdb/go_ibm_db@v0.2.0/odbcstmt.go:107 +0x94c
github.com/ibmdb/go_ibm_db.(*Stmt).Exec(0xc000298000, 0xc000322640, 0x14, 0x14, 0x0, 0x0, 0x0, 0x0)
        C:/Users/sdelacruz/Documents/go/pkg/mod/github.com/ibmdb/go_ibm_db@v0.2.0/stmt.go:66 +0xf0
database/sql.ctxDriverStmtExec(0xee74c0, 0xc000020108, 0xee8100, 0xc000298000, 0xc000017500, 0x14, 0x14, 0x14, 0x14, 0x0, ...)
        c:/go/src/database/sql/ctxutil.go:77 +0x16c
database/sql.resultFromStatement(0xee74c0, 0xc000020108, 0xee55c0, 0xc00029d3e0, 0xc00002c040, 0xc000322140, 0x14, 0x14, 0x0, 0x0, ...)
        c:/go/src/database/sql/sql.go:2435 +0x15d
database/sql.(*Stmt).ExecContext(0xc000318000, 0xee74c0, 0xc000020108, 0xc000322140, 0x14, 0x14, 0x0, 0x0, 0x0, 0x0)
        c:/go/src/database/sql/sql.go:2411 +0x200
        c:/go/src/database/sql/sql.go:2423
...
exit status 2
```

--- a/param.go
+++ b/param.go
@@ -119,17 +119,16 @@ func (p *Parameter) BindValue(h api.SQLHSTMT, idx int, v driver.Value) error {
 		}
 		size = 20 + api.SQLULEN(decimal)
 	case []byte:
-		lenD := len(d)
 		ctype = api.SQL_C_BINARY
-		b := make([]byte, lenD)
+		b := make([]byte, len(d))
 		copy(b, d)
 		p.Data = b
-		if lenD > 0 {
+		if len(d) > 0 {
 			buf = unsafe.Pointer(&b[0])
 		}
-		buflen = api.SQLLEN(lenD)
+		buflen = api.SQLLEN(len(b))
 		plen = p.StoreStrLen_or_IndPtr(buflen)
-		size = api.SQLULEN(lenD)
+		size = api.SQLULEN(len(b))
 		sqltype = api.SQL_BINARY
 	case sql.Out:
 		o, err := newOut(h, &d, idx)

--- a/param.go
+++ b/param.go
@@ -119,14 +119,17 @@ func (p *Parameter) BindValue(h api.SQLHSTMT, idx int, v driver.Value) error {
 		}
 		size = 20 + api.SQLULEN(decimal)
 	case []byte:
+		lenD := len(d)
 		ctype = api.SQL_C_BINARY
-		b := make([]byte, len(d))
+		b := make([]byte, lenD)
 		copy(b, d)
 		p.Data = b
-		buf = unsafe.Pointer(&b[0])
-		buflen = api.SQLLEN(len(b))
+		if lenD > 0 {
+			buf = unsafe.Pointer(&b[0])
+		}
+		buflen = api.SQLLEN(lenD)
 		plen = p.StoreStrLen_or_IndPtr(buflen)
-		size = api.SQLULEN(len(b))
+		size = api.SQLULEN(lenD)
 		sqltype = api.SQL_BINARY
 	case sql.Out:
 		o, err := newOut(h, &d, idx)

